test/checkout.int.t.sol

```diff
@@ -361,3 +368,139 @@ contract Checkout_buyingFromTenSuppliers is Checkout {
     );
   }
 }
+
+contract Checkout_buyingWithAlternateERC20 is Checkout {
+  MockERC20Permit internal _erc20;
+  SignatureUtils internal _mockERC20SignatureUtils;
+  uint256 deadline = 1 days;
+  uint256 ownerPrivateKey = 0xA11CE;
+  address owner = vm.addr(ownerPrivateKey);
+  uint256 amount;
+  uint256 certificateAmount;
+
+  function setUp() external {
+    _erc20 = _deployMockERC20();
+
+    _mockERC20SignatureUtils = new SignatureUtils(_erc20.DOMAIN_SEPARATOR());
+    _market.setPurchasingTokenAndPriceMultiple({
+      purchasingToken: _erc20,
+      priceMultiple: 2000
+    });
+    assertEq(_market.purchasingTokenAddress(), address(_erc20));
+    amount = _market.calculateCheckoutTotal(1 ether);
+    certificateAmount = _market.calculateCertificateAmountFromPurchaseTotal(
+      amount
+    );
+    _erc20.transfer(owner, amount);
+    assertEq(_erc20.balanceOf(address(owner)), amount);
+    _removalIds = _seedRemovals({
+      to: _namedAccounts.supplier,
+      count: 1,
+      list: true
+    });
+  }
+
+  function test() external {
+    assertEq(_removal.getMarketBalance(), 1 ether);
+    assertEq(_removal.numberOfTokensOwnedByAddress(address(_market)), 1);
+    _assertExpectedBalances(_namedAccounts.supplier, 0, false, 0);
+    _assertExpectedBalances(address(_certificate), 0, false, 0);
+    assertEq(_removal.balanceOf(address(_certificate), _removalIds[0]), 0);
+    vm.expectRevert(IERC721AUpgradeable.OwnerQueryForNonexistentToken.selector);
+    _certificate.ownerOf(_certificateTokenId);
+    SignedPermit memory signedPermit = _mockERC20SignatureUtils.generatePermit(
+      ownerPrivateKey,
+      address(_market),
+      amount,
+      1 days,
+      _erc20
+    );
+    vm.startPrank(owner);
+    _market.swap(
+      owner,
+      amount,
+      signedPermit.permit.deadline,
+      signedPermit.v,
+      signedPermit.r,
+      signedPermit.s
+    );
+    _assertExpectedBalances(address(_market), 0, false, 0);
+    _assertExpectedBalances(_namedAccounts.supplier, 0, false, 0);
+    _assertExpectedBalances(address(_certificate), certificateAmount, true, 1);
+    assertEq(
+      _removal.balanceOf(address(_certificate), _removalIds[0]),
+      certificateAmount
+    );
+    assertEq(_certificate.ownerOf(_certificateTokenId), owner);
+    vm.stopPrank();
```

Comment:
Would be useful to also test the new event you added here (or elsewhere)

Reply:
done